app/models/locations/map.rb

```diff
@@ -0,0 +1,90 @@
+module Locations
+  class Map
+    include ActiveModel::Model
+
+    def initialize(params = {})
```

Comment:
I think in general there's too much happening in initialization. Usually all you want to do here is take the arguments and set instance variables from those arguments which accessors will use. Things like `holding_location` can be instance methods which populate an instance variable on first access.